pavelib/assets.py

```diff
@@ -384,14 +384,24 @@ def coffeescript_files():
 
 @task
 @no_help
-def compile_coffeescript(*files):
+def compile_coffeescript(*coffee_files):
     """
     Compile CoffeeScript to JavaScript.
     """
-    if not files:
-        files = ["`{}`".format(coffeescript_files())]
+    js_files = ""
+
+    if not coffee_files:
+        coffee_files = ["`{}`".format(coffeescript_files())]
+        js_files = r"`{} | sed 's/\.coffee/\.js/g'`".format(coffeescript_files())
+    else:
+        js_files = r"`{} | sed 's/\.coffee/\.js/g'`".format(*coffee_files)
```

Comment:
Yes,,, sort of - `coffee_files` can be either a list of files, or a inlined bash command to find a list of files. You have to run it through bash first to normalize it to something you could run `re.sub` on first. So to rewrite this using `re.sub` instead of `sed` I'd need to make three separate calls of `sh(cmd())` - one to compile the coffeescript, one to turn `coffee_files` into something I can call re.sub on, and one to add the header to the JS files.

(You're definitely right about calling `coffeescript_files` more than once, though.)